Source/ARTRest.m

```diff
@@ -742,6 +742,19 @@ - (ARTLocalDevice *)device_nosync {
     return _sharedDevice;
 }
 
+/*
+ This function is needed to return functioning local device, since after `reset` call
+ you will end up with empty string as a device's `id` on the server.
+ */
+- (ARTLocalDevice *)loadDevice {
```

Comment:
Since this calls `device_noSync`, should this also have `noSync` in its name to indicate that it's the caller's responsibility to ensure that this method is called on the correct queue? (And then make sure that we respect this requirement in the places where we call this method.)